tests/unit/components/ContentNode/TabNavigator.spec.js

```diff
@@ -69,4 +69,49 @@ describe('TabNavigator.spec', () => {
     expect(wrapper.find('.tabs-content').text()).toBe('Second');
     expect(tabnav.props('value')).toEqual(titles[1]);
   });
+
+  it('selects correct tab when adding a tab', async () => {
+    const longerTitles = ['Long tab title',
+      'A Longer tab title', 'The Longest tab title', 'added title'];
+
+    const wrapper = createWrapper();
+    wrapper.setProps({ titles: longerTitles });
+
+    expect(wrapper.vm.currentTitle).toEqual('added title');
+  });
+
+  it('selects first tab when deleting current tab', async () => {
+    const longerTitles = ['Long tab title',
+      'A Longer tab title', 'The Longest tab title', 'added title'];
+
+    const wrapper = createWrapper();
+    // current tab is on 'added title'
+    wrapper.setProps({ titles: longerTitles });
+    wrapper.setProps({ titles });
+
+    expect(wrapper.vm.currentTitle).toEqual('Long tab title');
+  });
+
+  it('selects first tab when deleting current tab', async () => {
+    const longerTitles = ['Long tab title',
+      'A Longer tab title', 'The Longest tab title', 'added title'];
+    const removedTitles = ['Long tab title',
+      'A Longer tab title', 'added title'];
+    const wrapper = createWrapper();
+    // current tab is on 'added title'
+    wrapper.setProps({ titles: longerTitles });
+    wrapper.setProps({ titles: removedTitles });
+
+    expect(wrapper.vm.currentTitle).toEqual('added title');
+  });
+
+  it('selects correct tab when changing a tab', async () => {
+    const changedTitle = ['Long tab title',
+      'A Longer tab title', 'changed tab title'];
+
+    const wrapper = createWrapper();
+    wrapper.setProps({ titles: changedTitle });
+
+    expect(wrapper.vm.currentTitle).toEqual('changed tab title');
```

Comment:
You may want to test at least one other index (in the beginning or middle or both) to make sure your code is not just accidentally getting it right.